app/controllers/users_controller.rb

```diff
@@ -1,7 +1,66 @@
 class UsersController < ApplicationController
 
-  def list
+  def show
     @users = User.all
   end
-  alias :index :list
+  alias :index :show
+
+  def new
+  end
+
+  def create
+    @user = User.create(user_params)
+    redirect_to '/'
+  end
+
+  def trim
+    user = User.find(params[:id])
+    user.update_attribute(:trimmed, true)
+    user.save
+    render :json => user
+  end
+
+  def massage_and_trim
+    user = User.find(params[:id])
+    user.update_attributes(:trimmed => true, :massaged => true)
+    user.save
+    render :json => user
+  end
+
+  def shave
+    user = User.find(params[:id])
+    user.update_attribute(:shaved, true)
+    user.save
+    render :json => user
+  end
+
+  def reset
```

Comment:
I'm really baffled by the logic here, it is quite astonishing.  I would expect a 'reset' on a user to take them to everything being the initial state.  What this appears to do is...
If they are trimmed and massged, make them not trimmed and massaged.
If they are trimmed, make them not trimmed
if they are shaved, make them not shaved.

I am totally confused why it is implemented this way.